src/marqo/marqo_url_resolver.py

```diff
@@ -0,0 +1,30 @@
+import time
+
+import requests
+
+
+class MarqoUrlResolver:
+    def __init__(self, api_key=None, expiration_time: int = 60):
+        """ URL Resolver is a cache for urls that are resolved to their respective indices only for marqo cloud. """
+        self.timestamp = time.time() - expiration_time
+        self._urls_mapping = {}
+        self.api_key = api_key
+        self.expiration_time = expiration_time
+
+    def refresh_urls_if_needed(self, index_name):
+        if index_name not in self._urls_mapping and time.time() - self.timestamp > self.expiration_time:
```

Comment:
This basically means that we are OK with an index_name's url _never_ being updated in this client once set (including if it has been deleted and recreated). Are we sure we are alright with this?

Reply:
how about we refresh it once in 5-15 minutes if index is already presented, if customer were to delete and recreate their deletion would take around 5 minutes and after <=(5-15) minutes their index will be deleted from urls list and then each time they request it would refresh at the same rate as we have default.